src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -431,19 +482,53 @@ def _on_export_data_done(self, future):
         data = future.result()
         self._display_acquired_data(data)
 
-    def _export_data(self, data, thumb_nail):
+
+    def _create_cryo_filename(self, filename: str, acq_type: Optional[str] = None) -> str:
+        """
+        Create a filename for cryo images depending on mode.
+        :param filename: filename given by user
+        """
+        acq_map: Dict[str, str] = {
+            model.MD_AT_EM: "SEM",
+            model.MD_AT_FIB: "FIB"
+        }
+        if self.acqui_mode is guimod.AcquiMode.FLM:
+            filename = add_feature_info_to_filename(feature=self._tab_data.main.currentFeature.value,
+                                        filename=filename)
+        else:
+            filename = _create_fibsem_filename(filename, acq_map[acq_type])
+
+        return filename
+
+    def _export_data(self, data: List[model.DataArray], thumb_nail):
         """
         Called to export the acquired data.
         data (DataArray): the returned data/images from the future
         thumb_nail (DataArray): the thumbnail of the views
         """
-        filename = self._filename.value
-        if data:
-            filename = add_feature_info_to_filename(feature=self._tab_data.main.currentFeature.value,
-                                                 filename=filename)
+        auto_save: bool = True
```

Comment:
A global default variable?

Reply:
just a local one, we wanna update it everytime we export